apps/api/src/billing/repositories/usage/usage.repository.ts

```diff
@@ -0,0 +1,111 @@
+import { QueryTypes } from "sequelize";
+import { singleton } from "tsyringe";
+
+import { chainDb } from "@src/db/dbConnection";
+
+export interface BillingUsageRawResult {
+  date: string;
+  activeLeases: number;
+  dailyAktSpent: number;
+  totalAktSpent: number;
+  dailyUsdcSpent: number;
+  totalUsdcSpent: number;
+  dailyUsdSpent: number;
+  totalUsdSpent: number;
+}
+
+@singleton()
+export class UsageRepository {
+  async getHistory(address: string, startDate: string, endDate: string) {
+    const query = `
+      WITH date_range AS (
+        SELECT generate_series(
+          :startDate::date,
+          :endDate::date,
+          '1 day'::interval
+        )::date as date
+      ),
```

Comment:
issue(blocking): this SQL query contains the main logic responsible to calculating and gathering statistic, but there are no tests that covers it.

It's important to ensure this query properly work, otherwise it will be hard to change it